lib/appoptics_apm/oboe_init_options.rb

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2019 SolarWinds, LLC.
+# All rights reserved.
+
+require 'singleton'
+
+module AppOpticsAPM
+
+  class OboeInitOptions
+    include Singleton
+
+    attr_reader :reporter, :host
+
+    # TODO decide if these globals are useful when testing
+    # OBOE_HOSTNAME_ALIAS = 0
+    # OBOE_DEBUG_LEVEL = 1
+    # OBOE_LOGFILE = 2
+    #
+    # OBOE_MAX_TRANSACTIONS = 3
+    # OBOE_FLUSH_MAX_WAIT_TIME = 4
+    # OBOE_EVENTS_FLUSH_INTERVAL = 5
+    # OBOE_EVENTS_FLUSH_BATCH_SIZE = 6
+    #
+    # OBOE_REPORTER = 7
+    # OBOE_COLLECTOR = 8
+    # OBOE_SERVICE_KEY = 9
+    # OBOE_TRUSTEDPATH = 10
+    #
+    # OBOE_BUFSIZE = 11
+    # OBOE_TRACE_METRICS = 12
+    # OBOE_HISTOGRAM_PRECISION = 13
+    # OBOE_TOKEN_BUCKET_CAPACITY = 14
+    # OBOE_TOKEN_BUCKET_RATE = 15
+    # OBOE_FILE_SINGLE = 16
+
+    def initialize
+      # optional hostname alias
+      @hostname_alias = ENV['APPOPTICS_HOSTNAME_ALIAS'] || AppOpticsAPM::Config[:hostname_alias] || ''
+      # level at which log messages will be written to log file (0-6)
+      @debug_level = (ENV['APPOPTICS_DEBUG_LEVEL'] || AppOpticsAPM::Config[:debug_level] || 3).to_i
```

Comment:
this code is duplicated in config.rb.

Reply:
I know! 
not proud of it. 
It's ok for now, I'll dry out things when I would have a third instance.